src/python/modules/ensembl/compara/citest/tabletest.py

```diff
@@ -0,0 +1,245 @@
+        str_info=str_info+"|exp:"+str(int_variation)
+        result["info"]=str_info
+        result["table"]=str_table_name
+        result["type"]="nb_row"
+        self.test_results.append(result)
+
+
+    def _get_number_row(self,obj_connection_engine, str_table_name):
+        str_request="select count(*) from "+str_table_name+";"
+        obj_request=text(str_request)
+        obj_connection=obj_connection_engine.connect()
+        obj_result=obj_connection.execute(obj_request)
+        nbRow=0
+        for tpl_row in obj_result:
+            int_nb_row=tpl_row[0]
+        obj_connection.close()
+        return int_nb_row
+
+
+    def _get_autorised_variation_by_group(self,lst_info_group, lst_variations):
+            dic_variation={}
+            for dic_var in lst_variations:
+                dic_variation[dic_var["group"]]=dic_var["var"]
+            dic_result={}
+            for tpl_group in lst_info_group:
+                dic_result[tpl_group[1]]=0
+                if tpl_group[1] in dic_variation:
+                    int_nb_var=int(tpl_group[0]*dic_variation[tpl_group[1]])
+                    dic_result[tpl_group[1]]=int_nb_var
+            return dic_result
+
+
+    def _test_table_number_row_by_group(self, str_table_name, lst_groups):
+
+        for dic_group in lst_groups:
+            str_group_by=dic_group["name"]
+            lst_variations=dic_group["variations"]
+
+            #do the check for the given table
+            lst_result_ref=self._get_number_row_by_group(self.ref_connection_engine, str_table_name, str_group_by)
+            lst_result_tar=self._get_number_row_by_group(self.target_connection_engine, str_table_name, str_group_by)
+
+            ## test if the bunber of group are similar between the two datases
+            ## if not then failure and stop here the group by check
+            result={}
+            result["succes"]=0
+            if len(lst_result_ref) == len(lst_result_tar):
+                result["succes"]=1
+            str_info="grp:"+str_group_by+"|ref:"+str(len(lst_result_ref))+"|tar:"+str(len(lst_result_tar))
+            result["info"]=str_info
+            result["table"]=str_table_name
+            result["type"]="nb_row_all_grp"
+            self.test_results.append(result)
+            if result["succes"]==0:#if the number of group is not the same then there is no reason to continue
+                return
+
+            #load the number of autoiruied vairaiton (if group not describe in json then the variaiton expected is 0)
+            dic_autorised_variation=self._get_autorised_variation_by_group(lst_result_ref, lst_variations)
+
+            ### for aech group check that the numbers do nat variate more than autorised.
+            i=0
+            while i < len(lst_result_ref):
+                int_variation=dic_autorised_variation[lst_result_ref[i][1]]#get the aurrised variation for the group
```

Comment:
```suggestion
                int_variation=dic_authorised_variation[lst_result_ref[i][1]] #get the authorised variation for the group
```
Please replace 'autorised' with 'authorised' throughout the file